specification/metadata-3.2.md

```diff
@@ -45,151 +44,161 @@ of `METADATA LIST` MUST be as follows:
+for doing so. Such omissions MUST NOT be disclosed.
+
+*Errors*: `ERR_NOMATCHINGKEYS`
 
 ### METADATA SET
 
 This subcommand MUST set a required key to an optional value. If no value is
 given, the key is removed; otherwise, the value is assigned to the key. The
-response MUST be one `RPL_KEYVALUE` event and one `RPL_METADATAEND` event. The
-format of `METADATA SET` MUST be as follows:
+response MUST be one `RPL_KEYVALUE` event, representing what was actually
+stored by the server, and one `RPL_METADATAEND` event. The format of
+ `METADATA SET` MUST be as follows:
 
 `METADATA <Target> SET <Key> [:Value]`
 
-It is an error for users to set keys for other users, for services accounts
-other than their own (without an appropriate operator privilege), for channels
-for which they lack implementation-defined permission, or for
-implementation-defined keys which would require special access which the user
-does not possess.
+It is an error for users to set keys on targets for which they lack
+authorization from the server, and the server MUST respond with
+`ERR_KEYNOPERMISSION`.
 
-*Errors*: ERR_KEYINVALID, ERR_KEYNOTSET, ERR_KEYNOPERMISSION
+*Errors*: `ERR_METADATALIMIT`, `ERR_KEYINVALID`, `ERR_KEYNOTSET`, `ERR_KEYNOPERMISSION`
 
 ### METADATA CLEAR
 
-This subcommand MUST remove all metadata, equivalently to using METADATA SET
-on all keys with an empty value. The format of `METADATA CLEAR` MUST be as
-follows:
+This subcommand MUST remove all metadata, equivalently to using `METADATA SET`
+on all currently-set keys with an empty value. The format of `METADATA CLEAR`
+MUST be as follows:
 
 `METADATA <Target> CLEAR`
 
-This subcommand has the same conditions and errors as the `METADATA SET`
-subcommand.
+The server MUST respond with one `RPL_KEYVALUE` event per cleared key and one
+`RPL_METADATAEND` event. It is an error for users to use this subcommand on
+targets for which they lack authorization from the server. Servers MAY reject
+this subcommand for channels, using `ERR_KEYNOPERMISSION` with an asterisk
+(`*`) in the `<Key>` field.
 
 ## Metadata Notifications
 
-Notifications, e.g. from metadata subscriptions, MUST be passed to the client
-using the incoming METADATA verb, the format of which is as follows:
+Metadata notifications are enabled by requesting the `metadata-notify`
+capability during capability negotiation. When negotiated, this capability
+extends `MONITOR` behaviour to include subscribing users to notifications for
+those users they are currently monitoring. Clients are also subscribed to
+notifications for channels they join. Clients may discontinue notifications
+for users by issuing a disabling `MONITOR` command, and for channels by
+parting the channel. Clients are automatically subscribed to notifications for
+their own metadata.
+
+Notifications use the `METADATA` event, the format of which is as follows:
 
 `METADATA <Source> <Target> <Key> :<Value>`
```

Comment:
The source could be any entity capable of making a change, not just legal prefixes.